test/rage2e/rag_test.go

```diff
@@ -440,3 +447,118 @@ func deleteWorkspace(workspaceObj *kaitov1alpha1.Workspace) error {
 
 	return nil
 }
+
+func createIndexPod(ragengineObj *kaitov1alpha1.RAGEngine) error {
+	By("Creating index pod", func() {
+		pod := GenerateIndexPodManifest(ragengineObj.Namespace, ragengineObj.Name)
+		Eventually(func() error {
+			return utils.TestingCluster.KubeClient.Create(ctx, pod, &client.CreateOptions{})
+		}, utils.PollTimeout, utils.PollInterval).
+			Should(Succeed(), "Failed to create index pod")
+	})
+	time.Sleep(60 * time.Second)
+
+	return nil
+}
+
+func createAndValidateQueryPod(ragengineObj *kaitov1alpha1.RAGEngine) error {
+	By("Creating query pod", func() {
+		pod := GenerateQueryPodManifest(ragengineObj.Namespace, ragengineObj.Name)
+		Eventually(func() error {
+			return utils.TestingCluster.KubeClient.Create(ctx, pod, &client.CreateOptions{})
+		}, utils.PollTimeout, utils.PollInterval).
+			Should(Succeed(), "Failed to create query pod")
+	})
+	time.Sleep(30 * time.Second)
+	By("Checking the query logs", func() {
+		Eventually(func() bool {
+			coreClient, err := utils.GetK8sClientset()
+			if err != nil {
+				GinkgoWriter.Printf("Failed to create core client: %v\n", err)
+				return false
+			}
+
+			logs, err := utils.GetPodLogs(coreClient, ragengineObj.Namespace, "querypod", "")
+			if err != nil {
+				GinkgoWriter.Printf("Failed to get logs from pod %s: %v\n", "querypod", err)
+				return false
+			}
+
+			searchQuerySuccess := "'text': '\\\\nKaito is an operator that automates the AI/ML model inference or tuning workload in a Kubernetes cluster.\\\\n"
```

Comment:
Shouldn't we check the result of the request? Check the request here?

Reply:
I think this is the request checking